.github/CONTRIBUTING.md

```diff
@@ -81,6 +99,20 @@ $ make html
 
 The built documentation can then be found in `docs/_build/html/`.
 
+To file a pull request, create a new branch on top of the upstream repository:
+
+```console
+$ git checkout -b my_topical_branch upstream/main
+```
+
+Make your changes, push them to your fork (the remote *origin*) and publish the PR!
+
+Before starting to work on your next pull request, run the following command to sync your local repository with the remotes:
+
+```console
+$ git fetch --all
```

Comment:
```suggestion
$ git fetch --all
$ git checkout main
$ git merge
```

Reply:
Which `main` is this pointing at? I'll depend on whether the repo has their local branch in sync with upstream or fork, or is outdated.

Reply:
Not exactly. The connection between a local branch and a remote tracking branch lives in `.git/config`. If it's changed, it may end up merging a remote branch from a different remote than what you might expect. Though, if one were to follow this guide from the beginning (and not change anything in their Git config), I'm almost sure it won't break the assumptions... I actually don't remember ever using `git merge` without the args.

OTOH, assuming that a contributor does not touch their local `main` branch that is configured with `upstream/main` as tracking, I'd replace these three commands with a `git pull --rebase` or `git checkout main && git fetch --all && git reset upstream/main` if the goal is to make the local `main` point to the same commit as the upstream one.

Yet, honestly, I'd only leave the `git fetch --all` command that would sync the remote tree cache and not worry about telling people how to change their local `main`. It's not necessary to do this for making new contributions, given that they follow the instructions of creating their topic branches on top of `upstream/main` and rebasing on top of it, if the need be.

Reply:
> Yet, honestly, I'd only leave the `git fetch --all` command that would sync the remote tree cache and not worry about telling people how to change their local `main`. It's not necessary to do this for making new contributions, given that they follow the instructions of creating their topic branches on top of `upstream/main` and rebasing on top of it, if the need be.

I also think this is the most efficient & simple solution.